Flagging this build error on the side: Something went wrong in the way things got rebased and `git --describe` no longer works. Merging this would introduce a wrong tag. This confirms that assertions and exceptions are raised with good reasons @indirectlylit @rtibbles :)

I still prefer to stick to the "paranoid" handling of `git --describe`, until all unexpected behavior is accounted for :)

```
AssertionError: Version detected from git describe --tags, but it's inconsistent with kolibri.__version__.__version__ is: (0, 10, 0, u'beta', 1), tag says: (0, 11, 0, 'alpha', 1).
```

Referring to #3678

--- a/kolibri/utils/options.py
+++ b/kolibri/utils/options.py
@@ -33,10 +33,10 @@ option_spec = {
         },
     },
     "Server": {
-        "THREADS_POOL": {
+        "THREAD_POOL": {
             "type": "integer",
             "default": 10,
-            "envvars": ("CHERRYPY_THREADS_POOL",),
+            "envvars": ("CHERRYPY_THREAD_POOL",),
         },
         "SOCKET_TIMEOUT": {
             "type": "integer",

--- a/kolibri/utils/options.py
+++ b/kolibri/utils/options.py
@@ -32,6 +32,28 @@ option_spec = {
             "envvars": ("KOLIBRI_DATABASE_HOST",),
         },
     },
+    "Server": {
+        "THREADS_POOL": {
+            "type": "integer",
+            "default": 10,
+            "envvars": ("CHERRYPY_THREADS_POOL",),
+        },
+        "SOCKET_TIMEOUT": {
+            "type": "integer",
+            "default": 10,
+            "envvars": ("CHERRYPY_SOCKET_TIMEOUT",),
+        },
+        "QUEUE_SIZE": {
+            "type": "integer",
+            "default": 5,
+            "envvars": ("CHERRYPY_QUEUE_SIZE",),
+        },
+        "QUEUE_TIMEOUT": {
+            "type": "float",
+            "default": 0.1,
+            "envvars": ("CHERRYPY_QUEUE_TIMEOUT",),
+        },
+    },
     "Paths": {
         "CONTENT_DIR": {
             "type": "string",

--- a/kolibri/utils/options.py
+++ b/kolibri/utils/options.py
@@ -33,25 +33,25 @@ option_spec = {
         },
     },
     "Server": {
-        "THREAD_POOL": {
+        "SERVER_THREAD_POOL": {
             "type": "integer",
             "default": 10,
-            "envvars": ("CHERRYPY_THREAD_POOL",),
+            "envvars": ("KOLIBRI_SERVER_THREAD_POOL",),
         },
-        "SOCKET_TIMEOUT": {
+        "SERVER_SOCKET_TIMEOUT": {
             "type": "integer",
             "default": 10,
-            "envvars": ("CHERRYPY_SOCKET_TIMEOUT",),
+            "envvars": ("KOLIBRI_SERVER_SOCKET_TIMEOUT",),
         },
-        "QUEUE_SIZE": {
+        "SERVER_QUEUE_SIZE": {
             "type": "integer",
             "default": 5,
-            "envvars": ("CHERRYPY_QUEUE_SIZE",),
+            "envvars": ("KOLIBRI_SERVER_QUEUE_SIZE",),
         },
-        "QUEUE_TIMEOUT": {
+        "SERVER_QUEUE_TIMEOUT": {
             "type": "float",
             "default": 0.1,
-            "envvars": ("CHERRYPY_QUEUE_TIMEOUT",),
+            "envvars": ("KOLIBRI_SERVER_QUEUE_TIMEOUT",),
         },
     },
     "Paths": {

--- a/kolibri/utils/server.py
+++ b/kolibri/utils/server.py
@@ -151,10 +151,10 @@ def run_server(port):
     # Configure the server
     server.socket_host = LISTEN_ADDRESS
     server.socket_port = port
-    server.thread_pool = conf.OPTIONS["Server"]["THREAD_POOL"]
-    server.socket_timeout = conf.OPTIONS["Server"]["SOCKET_TIMEOUT"]
-    server.accepted_queue_size = conf.OPTIONS["Server"]["QUEUE_SIZE"]
-    server.accepted_queue_timeout = conf.OPTIONS["Server"]["QUEUE_TIMEOUT"]
+    server.thread_pool = conf.OPTIONS["Server"]["SERVER_THREAD_POOL"]
+    server.socket_timeout = conf.OPTIONS["Server"]["SERVER_SOCKET_TIMEOUT"]
+    server.accepted_queue_size = conf.OPTIONS["Server"]["SERVER_QUEUE_SIZE"]
+    server.accepted_queue_timeout = conf.OPTIONS["Server"]["SERVER_QUEUE_TIMEOUT"]
 
     # Subscribe this server
     server.subscribe()

--- a/kolibri/utils/server.py
+++ b/kolibri/utils/server.py
@@ -151,7 +151,7 @@ def run_server(port):
     # Configure the server
     server.socket_host = LISTEN_ADDRESS
     server.socket_port = port
-    server.thread_pool = conf.OPTIONS["Server"]["THREADS_POOL"]
+    server.thread_pool = conf.OPTIONS["Server"]["THREAD_POOL"]
     server.socket_timeout = conf.OPTIONS["Server"]["SOCKET_TIMEOUT"]
     server.accepted_queue_size = conf.OPTIONS["Server"]["QUEUE_SIZE"]
     server.accepted_queue_timeout = conf.OPTIONS["Server"]["QUEUE_TIMEOUT"]

--- a/kolibri/utils/server.py
+++ b/kolibri/utils/server.py
@@ -12,7 +12,7 @@ from django.core.management import call_command
 from .system import kill_pid
 from .system import pid_exists
 from kolibri.content.utils import paths
-from kolibri.utils.conf import KOLIBRI_HOME
+from kolibri.utils import conf
 
 logger = logging.getLogger(__name__)
 
@@ -30,15 +30,15 @@ STATUS_PID_FILE_INVALID = 100
 STATUS_UNKNOWN = 101
 
 # Used to store PID and port number (both in foreground and daemon mode)
-PID_FILE = os.path.join(KOLIBRI_HOME, "server.pid")
+PID_FILE = os.path.join(conf.KOLIBRI_HOME, "server.pid")
 
 # Used to PID, port during certain exclusive startup process, before we fork
 # to daemon mode
-STARTUP_LOCK = os.path.join(KOLIBRI_HOME, "server.lock")
+STARTUP_LOCK = os.path.join(conf.KOLIBRI_HOME, "server.lock")
 
 # This is a special file with daemon activity. It logs ALL stderr output, some
 # might not have made it to the log file!
-DAEMON_LOG = os.path.join(KOLIBRI_HOME, "server.log")
+DAEMON_LOG = os.path.join(conf.KOLIBRI_HOME, "server.log")
 
 # Currently non-configurable until we know how to properly handle this
 LISTEN_ADDRESS = "0.0.0.0"
@@ -151,7 +151,10 @@ def run_server(port):
     # Configure the server
     server.socket_host = LISTEN_ADDRESS
     server.socket_port = port
-    server.thread_pool = 30
+    server.thread_pool = conf.OPTIONS["Server"]["THREADS_POOL"]
+    server.socket_timeout = conf.OPTIONS["Server"]["SOCKET_TIMEOUT"]
+    server.accepted_queue_size = conf.OPTIONS["Server"]["QUEUE_SIZE"]
+    server.accepted_queue_timeout = conf.OPTIONS["Server"]["QUEUE_TIMEOUT"]
 
     # Subscribe this server
     server.subscribe()

